go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -410,20 +410,18 @@ func (agent *ActionAgent) setMasterLocked(ctx context.Context, parentAlias *topo
 		return err
 	}
 
-	// If this tablet used to be a master, end orchestrator maintenance after we are connected to the new master.
+	// End orchestrator maintenance at the end of fixing replication.
 	// This is a best effort operation, so it should happen in a goroutine
-	if agent.Tablet().Type == topodatapb.TabletType_MASTER {
-		defer func() {
-			go func() {
-				if agent.orc == nil {
-					return
-				}
-				if err := agent.orc.EndMaintenance(agent.Tablet()); err != nil {
-					log.Warningf("Orchestrator EndMaintenance failed: %v", err)
-				}
-			}()
+	defer func() {
```

Comment:
This is perhaps not super clear. I'm removing the conditional ```if agent.Tablet().Type == topodatapb.TabletType_MASTER``` here as, in the case where the tablet was a REPLICA, and we're calling into this function from repairReplication we also want to end Orchestrator maintenance at the end of this function. It's still best effort to end orchestrator maintenance even in that case, as Orchestrator maintenance windows are time bounded.

Reply:
I think this is ok for now. Ideally, whoever calls begins maintenance should be the one ending it (in a defer).
So, in the case of `repairReplication`, the block that calls the `beginMaintenance` should defer `endMaintenance`.
But then, we should also move other call sites to the vtctld workflow (wrangler), where begin and end should be called as part of the reparent commands. This probably means that vtctld needs to know about Orc. So, this may be a more involved change.

Reply:
I agree, that sounds a lot better. I'll do that refactoring in a quick followup to this PR.